strings/hamming_distance.rb

```diff
@@ -0,0 +1,23 @@
+# https://en.wikipedia.org/wiki/Hamming_distance
+
+def hamming_distance(str1, str2)
+  abort 'Strings must be of the same length' unless str1.length == str2.length
+
+  str1.chars.zip(str2.chars).sum { |chr1, chr2| chr1 == chr2 ? 0 : 1 }
```

Comment:
Note: I think this code can be written a bit shorter as follows:

```suggestion
  str1.chars.zip(str2.chars).count { |chr1, chr2| chr1 != chr2 }
```